tests/test_pad.py

```diff
@@ -0,0 +1,123 @@
+import copy
+import os
+
+import numpy as np
+import pytest
+import yaml
+from pytest_check import check
+
+from modsim2.data.loader import DMPair
+from modsim2.similarity.metrics import pad
+
+
+# Fixture that returns the metric config dictionary
+@pytest.fixture(scope="module")
+def metrics_config() -> dict:
+    project_root = os.path.abspath(os.path.join(os.path.dirname(__file__), os.pardir))
+    metrics_config_path = os.path.join(
+        project_root, "tests", "testconfig", "metrics.yaml"
+    )
+    with open(metrics_config_path, "r") as stream:
+        pad_config = yaml.safe_load(stream)
+    # filter down to only pad configs
+    pad_config = {k: v for k, v in pad_config.items() if v["class"] == "pad"}
+    return pad_config
+
+
+# This test checks that the distance between a dataset and itself is the expect value
+# For this metric the expected value is always zero
+def test_cifar_pad_same(metrics_config: dict):
+    dmpair = DMPair(metrics_config=metrics_config)
+    similarity_dict = dmpair.compute_similarity()
+    similarity_dict_only_train = dmpair.compute_similarity(only_train=True)
+    test_scenarios = {
+        "same_result": similarity_dict,
+        "same_result_only_train": similarity_dict_only_train,
+    }
+    compare_results(test_scenarios, metrics_config)
+
+
+# This test checks that the distance between two different datasets is the expected
+# value
+# The calculated distance is checked against the known value for the seed - brittle
+# test but useful for messing with code
+def test_cifar_pad_different(metrics_config: dict):
+    dmpair = DMPair(metrics_config=metrics_config, drop_percent_A=0.2, seed=42)
+    similarity_dict = dmpair.compute_similarity()
+    similarity_dict_only_train = dmpair.compute_similarity(only_train=True)
+    test_scenarios = {
+        "diff_result": similarity_dict,
+        "diff_result_only_train": similarity_dict_only_train,
+    }
+    compare_results(test_scenarios, metrics_config)
+
+
+# This test is a sanity check that ensures the PAD between two datasets
+# with distinct labels is greater than when the datasets labels contain
+# the same classes. In theory the classifiers should be more able to
+# distinguish between the two datasets if their classes are distinct.
+def test_cifar_pad_distinct(metrics_config: dict):
+
+    dmpair = DMPair(metrics_config=metrics_config)
+    similarity_dict = dmpair.compute_similarity()
+
+    metrics_config_copy = copy.deepcopy(metrics_config)
+    for k in metrics_config_copy:
+        metrics_config_copy[k]["arguments"]["distinct_classes"] = True
+    dmpair = DMPair(metrics_config=metrics_config_copy)
+    similarity_dict_distinct_classes = dmpair.compute_similarity()
+
+    for k in metrics_config:
+        assert similarity_dict[k] < similarity_dict_distinct_classes[k]
+
+
+# This test checks that there are equal samples of A and B in the
+# combined test dataset
+def test_cifar_pad_equal_samples(metrics_config: dict):
```

Comment:
is the aim here to check your `__train_test_split` logic?

Reply:
tricky one. I think it's fine to create a PAD object here. I think there's probably then a question of why we don't always do this for the metrics (are we testing `compute_similarity` or `calculate_distance`? but I think ok for our purposes in this repo.

I think using a spy (like a mock) here may be useful to see what's been passed to methods (to examine test labels) rather than accessing via properties on the PAD object (have left a separate comment about removing unused properties)
I _think_ the best way to create a spy is with the `wraps` param on the mock: https://docs.python.org/3/library/unittest.mock.html (or similar in `patch`)
(assuming changes to `__evaluate_models` to pass in kwarg `labels` rather than using properties on `self` (if that change occurs)
``` from unittest.mock import patch```
```
        pad_metric = pad.PAD(seed=dmpair.seed)
        with patch.object(
            pad_metric, "_evaluate_models", wraps=pad_metric._evaluate_models
        ) as wrapped_evaluate_models:

            _ = pad_metric.calculate_distance(
                data_A=data_A,
                data_B=data_B,
                labels_A=labels_A,
                labels_B=labels_B,
                **metric["arguments"],
            )
            combined_test_labels = wrapped_calc_distance.call_args.kwargs["labels"]
            sum_test_labels = np.sum(combined_test_labels)
            count_test_labels = combined_test_labels.shape[0]
```

sorry, that's a bit messy to explain. Happy to chat.